capsules/src/rng.rs

```diff
@@ -186,3 +197,304 @@ impl<RNG: rng::RNG> Driver for SimpleRng<'a, RNG> {
         }
     }
 }
+
+pub struct Entropy32ToRandom<'a> {
+    egen: &'a Entropy32<'a>,
+    client: OptionalCell<&'a rng::Client>,
+}
+
+impl Entropy32ToRandom<'a> {
+    pub fn new(egen: &'a Entropy32<'a>) -> Entropy32ToRandom<'a> {
+        Entropy32ToRandom {
+            egen: egen,
+            client: OptionalCell::empty(),
+        }
+    }
+}
+
+impl Rng<'a> for Entropy32ToRandom<'a> {
+    fn get(&self) -> ReturnCode {
+        self.egen.get()
+    }
+
+    fn cancel(&self) -> ReturnCode {
+        self.egen.cancel()
+    }
+
+    fn set_client(&'a self, client: &'a rng::Client) {
+        self.egen.set_client(self);
+        self.client.set(client);
+    }
+}
+
+impl<'a> entropy::Client32 for Entropy32ToRandom<'a> {
+    fn entropy_available(
+        &self,
+        entropy: &mut Iterator<Item = u32>,
+        error: ReturnCode,
+    ) -> entropy::Continue {
+        self.client.map_or(entropy::Continue::Done, |client| {
+            if error != ReturnCode::SUCCESS {
+                match client.randomness_available(&mut Entropy32ToRandomIter(entropy), error) {
+                    rng::Continue::More => entropy::Continue::More,
```

Comment:
Consider implementing `From` or `Into` trait for ```rng::Continue``` and ```entropy::Continue``` enum instead of these manual conversations

Reply:
I thought about this -- I thought it better to have this simple conversion in-band. I don't want there to be general conversions possible, or have the conversion off somewhere else in the file.